### PR TITLE
Improve code coverage for scheduler/algorithm/priorities

### PR DESCRIPTION
**Pull request by tanshanshan**
**What this PR does / why we need it**:

Improve code coverage for scheduler/algorithm/priorities from #39559 

This is my first unit test  for kubernetes , thanks for your review.

**Special notes for your reviewer**:

**Release note**:

```release-note
```

### CODE REVIEW

**Comment by k82cn** [comment]
@k8s-bot ok to test

**Comment by k82cn** [comment]
@tanshanshan , please update BUILD file by `./hack/update-bazel.sh`.

File: plugin/pkg/scheduler/algorithm/priorities/metadata_test.go [hunk 1]
```diff
@@ -0,0 +1,84 @@
...
+limitations under the License.
+*/
+
+package priorities
+
+import (
+	"encoding/json"
+	"testing"
+
+	assert "github.com/stretchr/testify/assert"
```
Comment (k82cn) [inline plugin/pkg/scheduler/algorithm/priorities/metadata_test.go]: Let's just use `t.Errorf` for now.
There's several discussion on whether introduce some UT libs, but please do not introduce it in a single PR.

File: plugin/pkg/scheduler/algorithm/priorities/metadata_test.go [hunk 2]
```diff
@@ -0,0 +1,84 @@
...
+					Image:           "image",
+					ImagePullPolicy: "Always",
+				},
+			},
+		},
+	}
+
+	tests := []struct {
+		pod    *v1.Pod
+		test   string
```
Comment (k82cn) [inline plugin/pkg/scheduler/algorithm/priorities/metadata_test.go]: filed is not used?

File: plugin/pkg/scheduler/algorithm/priorities/metadata_test.go [hunk 3]
```diff
@@ -0,0 +1,84 @@
...
+		pMdata interface{}
```
Comment (k82cn) [inline plugin/pkg/scheduler/algorithm/priorities/metadata_test.go]: let's use some meaningful words, e.g. `expected`.

File: plugin/pkg/scheduler/algorithm/priorities/metadata_test.go [hunk 4]
```diff
@@ -0,0 +1,84 @@
...
+				nonZeroRequest: nonZeroReqs,
+				podTolerations: tolerations,
+				affinity:       nil,
+			},
+		},
+	}
+	for _, test := range tests {
+		ptData := PriorityMetadata(test.pod, nil)
+		assert.Equal(t, ptData, test.pMdata)
+
```
Comment (k82cn) [inline plugin/pkg/scheduler/algorithm/priorities/metadata_test.go]: un-neccessary empty line.

File: plugin/pkg/scheduler/algorithm/priorities/metadata_test.go [hunk 5]
```diff
@@ -0,0 +1,84 @@
...
+	}}
+	tolerationData, _ := json.Marshal(tolerations)
+	podWithTolerations := &v1.Pod{
+		ObjectMeta: v1.ObjectMeta{
+			Annotations: map[string]string{
+				v1.TolerationsAnnotationKey: string(tolerationData),
+			},
+		},
+		Spec: v1.PodSpec{
+			Containers: []v1.Container{
```
Comment (k82cn) [inline plugin/pkg/scheduler/algorithm/priorities/metadata_test.go]: Add a case for no zero request of container.

File: plugin/pkg/scheduler/algorithm/priorities/metadata_test.go [hunk 6]
```diff
@@ -0,0 +1,84 @@
...
+	"testing"
+
+	assert "github.com/stretchr/testify/assert"
+	"k8s.io/kubernetes/pkg/api/v1"
+	priorityutil "k8s.io/kubernetes/plugin/pkg/scheduler/algorithm/priorities/util"
+	"k8s.io/kubernetes/plugin/pkg/scheduler/schedulercache"
+)
+
+func TestPriorityMetadata(t *testing.T) {
+
```
Comment (k82cn) [inline plugin/pkg/scheduler/algorithm/priorities/metadata_test.go]: un-neccessary empty line.

File: plugin/pkg/scheduler/algorithm/priorities/metadata_test.go [hunk 7]
```diff
@@ -0,0 +1,84 @@
+/*
+Copyright 2016 The Kubernetes Authors.
```
Comment (k82cn) [inline plugin/pkg/scheduler/algorithm/priorities/metadata_test.go]: s/2016/2017

File: plugin/pkg/scheduler/algorithm/priorities/metadata_test.go [hunk 8]
```diff
@@ -0,0 +1,84 @@
...
+package priorities
+
+import (
+	"encoding/json"
+	"testing"
+
+	assert "github.com/stretchr/testify/assert"
+	"k8s.io/kubernetes/pkg/api/v1"
+	priorityutil "k8s.io/kubernetes/plugin/pkg/scheduler/algorithm/priorities/util"
+	"k8s.io/kubernetes/plugin/pkg/scheduler/schedulercache"
```
Comment (k82cn) [inline plugin/pkg/scheduler/algorithm/priorities/metadata_test.go]: please run `gofmt -w -s`

File: plugin/pkg/scheduler/algorithm/priorities/metadata_test.go [hunk 9]
```diff
@@ -0,0 +1,120 @@
...
+			pod:      nil,
+			expected: nil,
+			test:     "pod is nil , priorityMetadata is nil",
+		},
+		{
+			pod: podWithTolerations,
+			expected: &priorityMetadata{
+				nonZeroRequest: nonZeroReqs,
+				podTolerations: tolerations,
+				affinity:       nil,
```
Comment (timothysc) [inline plugin/pkg/scheduler/algorithm/priorities/metadata_test.go]: You should probably add an affinity check as well.

**Review by timothysc — CHANGES_REQUESTED** [review]
Please add a minor check on affinity too then lgtm

**Comment by timothysc** [comment]
@tanshanshan looks like you will need to update your bazel files.  

run ./hack/update-bazel.sh

**Comment by tanshanshan** [comment]
@timothysc 
I have run hack/update-bazel.sh , and update the plugin/pkg/scheduler/algorithm/priorities/BUILD

but , Jenkins Bazel Build failed.

**Comment by timothysc** [comment]
@k8s-bot bazel test this

**Comment by k82cn** [comment]
@tanshanshan , please fix the following error because of api-machinery refactor.

```
I0123 14:21:31.715] k8s.io/kubernetes/plugin/pkg/scheduler/algorithm/priorities/metadata_test.go:73: cannot use "k8s.io/kubernetes/pkg/api/v1".ObjectMeta literal (type "k8s.io/kubernetes/pkg/api/v1".ObjectMeta) as type "k8s.io/apimachinery/pkg/apis/meta/v1".ObjectMeta in field value
I0123 14:21:31.722] k8s.io/kubernetes/plugin/pkg/scheduler/algorithm/priorities/metadata_test.go:90: cannot use "k8s.io/kubernetes/pkg/api/v1".ObjectMeta literal (type "k8s.io/kubernetes/pkg/api/v1".ObjectMeta) as type "k8s.io/apimachinery/pkg/apis/meta/v1".ObjectMeta in field value
```

**Review by timothysc — APPROVED** [review]

**Pull request merged**